src/ResourceManager/AnalysisServices/Commands.AnalysisServices.ServiceManagement/Models/AsAzureClientSession.cs

```diff
@@ -0,0 +1,142 @@
+
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using System;
+using System.Security;
+
+namespace Microsoft.Azure.Commands.AnalysisServices.ServiceManagement
+{
+    /// <summary>
+    /// Represents current AS Azure session.
+    /// </summary>
+    public class AsAzureClientSession
+    {
+        public const string RestartEndpointPathFormat = "/webapi/servers/{0}/restart?api-version=2016-10-01";
+        public const string AadAuthorityUrlProd = "https://login.windows.net";
+        public const string AadAuthorityUrlPpe = "https://login.windows-ppe.net";
+        public const string AsAzureClientId = "cf710c6e-dfcc-4fa8-a093-d47294e44c66";
+        public static readonly Uri RedirectUri = new Uri("urn:ietf:wg:oauth:2.0:oob");
+
+        /// <summary>
+        /// Gets or sets the token cache store.
+        /// </summary>
+        public static TokenCache TokenCache { get; set; }
+        
+        /// <summary>
+        /// As Azure Profile
+        /// </summary>
+        private AsAzureProfile _profile;
+
+        static AsAzureClientSession()
+        {
+            Instance = new AsAzureClientSession();
+        }
+
+        private AsAzureClientSession()
+        {
+            TokenCache = new TokenCache();
+            _profile = new AsAzureProfile();
+        }
+
+        public static AsAzureClientSession Instance { get; private set; }
+
+        public AsAzureProfile Profile
+        {
+            get { return _profile; }
+            set
+            {
+                _profile = value;
+            }
+        }
+        
+        public AsAzureProfile Login(AsAzureContext asAzureContext, SecureString password)
+        {
+            PromptBehavior promptBehavior = password == null ? PromptBehavior.Always : PromptBehavior.Auto;
+
+            var resourceUri = new UriBuilder(Uri.UriSchemeHttps, asAzureContext.Environment.Name).ToString();
+            resourceUri = resourceUri.TrimEnd('/');
+            GetAadAuthenticatedToken(asAzureContext, password, promptBehavior, AsAzureClientId, resourceUri, RedirectUri);
+
+            _profile.Context.TokenCache = AsAzureClientSession.TokenCache.Serialize();
+
+            if (!_profile.Environments.ContainsKey(asAzureContext.Environment.Name))
+            {
+                _profile.Environments.Add(asAzureContext.Environment.Name, asAzureContext.Environment);
+            }
+
+            return _profile;
+        }
+
+        /// <summary>
+        /// Function to get an AAD token for a given user, client id and resource
+        /// </summary>
+        public static string GetAadAuthenticatedToken(AsAzureContext asAzureContext, SecureString password, PromptBehavior promptBehavior, string clientId, string resourceUri, Uri resourceRedirectUri)
+        {
+            var authUriBuilder = new UriBuilder(asAzureContext.Environment.Endpoints[AsAzureEnvironment.AsRolloutEndpoints.AdAuthorityBaseUrl]);
```

Comment:
Both of those options require the user to do Login-AzureRmAccount. For this dataplane api that does not work because the token audience is different. We discussed about this and agreed that I'll have to get token on my own.